pkg/controllers/reconciler.go

```diff
@@ -159,6 +187,47 @@ func (r *reconciler) createObjects(ctx context.Context,
 	return result, nil
 }
 
+func imageGetVersion(image string) string {
+	version := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		version = parts[1]
+	}
+
+	return version
+}
+
+func imageGetName(image string) string {
+	name := ""
+
+	parts := strings.SplitN(filepath.Base(image), ":", 2)
+
+	if len(parts) == 2 {
+		name = parts[0]
+	}
+
+	return name
+}
+
+func UpgradeToImage(image string) string {
+	out := ""
+
+	if len(UpgradeToVersion) > 0 {
+		out = imageGetName(image) + UpgradeToVersion
```

Comment:
The syntax for `UpgradeToVersion` is not checked. It should be done already during startup that we get to notify user as soon as possible.